# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1429293324 -7200
#      Fri Apr 17 19:55:24 2015 +0200
# Node ID 23fb65b45d8cd51a281d3a64cabcb3bc873426c9
# Parent  c58cf3a6551934a8f4fb90b4fc977f915e6f602d
do not call custom editor at startup and when debugging (bug #44701)

* file-editor.cc (call_custom_editor): return with true but without opening
  a file;
  (empty_script): do not open an empty script in the cutom editor
  at startup

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -267,16 +267,19 @@ file_editor::find_tab_widget (const QStr
 bool
 file_editor::call_custom_editor (const QString& file_name, int line)
 {
   // Check if the user wants to use a custom file editor.
   QSettings *settings = resource_manager::get_settings ();
 
   if (settings->value ("useCustomFileEditor",false).toBool ())
     {
+      if (line > -1)  // check for a specific line (debugging)
+        return true;  // yes: do ont open a file in external editor
+
       QString editor = settings->value ("customFileEditor").toString ();
       editor.replace ("%f", file_name);
       editor.replace ("%l", QString::number (line));
 
       bool started_ok = QProcess::startDetached (editor);
 
       if (started_ok != true)
         {
@@ -1911,16 +1914,20 @@ file_editor::check_actions ()
 
 // empty_script determines whether we have to create an empty script
 // 1. At startup, when the editor has to be (really) visible
 //    (Here we can not use the visibility changed signal)
 // 2. When the editor becomes visible when octave is running
 void
 file_editor::empty_script (bool startup, bool visible)
 {
+  QSettings *settings = resource_manager::get_settings ();
+  if (settings->value ("useCustomFileEditor",false).toBool ())
+    return;  // do not open an empty script in the external editor
+
   bool real_visible;
 
   if (startup)
     real_visible = isVisible ();
   else
     real_visible = visible;
 
   if (! real_visible || _tab_widget->count () > 0)
